extensions/rich_text_components/base.py

```diff
@@ -49,6 +49,12 @@ class BaseRichTextComponent(object):
     # Whether the component requires the filesystem in some way that
     # prevents it from being used by unauthorized users.
     requires_fs = False
+    # Whether the component should be displayed as a block element
+    is_block_element = False
+    # URL template of the component, such that when interpolated with the
+    # customization arugment dictionary, points to an image which will be
+    # rendered in the RTE preview. explorationId may be used as well.
+    preview_url_template = ''
```

Comment:
The way this is handled is still a bit convoluted, and it would be nicer to encapsulate all the "is there an image file? if so, do this, if not do that" stuff within here. So, how about making this a `@property` instead?

```
    @property
    def preview_url_template(self):
        return os.path.join(...)
```

Then, overwrite this for Image and Video. In addition, create preview pngs for Link and Math (that are exactly the same as the toolbar icons). I think that would be cleaner and would simplify the logic below, as well as in the frontend (there shouldn't need to be an else condition).